### DISCUSSION TITLE
Allow autogenerate sidebar from Content Layer API

### DISCUSSION THREAD

**Discussion by sergeysova** [withastro/starlight#2268] [Feature Requests]
### What version of `starlight` are you using?

0.26.1

### What is your idea?

It will be very useful to load items from [Content Layer API](https://docs.astro.build/en/reference/configuration-reference/#querying-and-rendering-with-the-content-layer-api) collection.

Like this:

```mjs
export default defineConfig({
  integrations: [
    starlight({
      sidebar: [
        {
          label: "Guides",
          autogenerate: { directory: "guides" },
        },
        {
          label: "Reference",
          items: [
            {
              label: "Operators",
              autogenerate: { collection: 'operators', filters },
            },
          ],
        },
      ],
    }),
  ],
  experimental: {
    contentLayer: true,
  },
});
```

Related discussion: https://github.com/withastro/roadmap/issues/946

### Why is this feature necessary?

This feature simplifies documentation maintenance by allowing developers to keep docs close to the code while still integrating them into the sidebar. It reduces fragmentation, improves consistency, and makes it easier to track what's documented.

## Example scenario:

```
documentation/
src/
├── method-A/
│   ├── index.ts
│   ├── methodA.test.ts
│   ├── readme.md
│   └── methodA.type-test.ts
├── method-B/
│   ├── index.ts
│   ├── methodB.test.ts
│   ├── readme.md
│   └── methodB.type-test.ts
└── ...
```

With this feature, docs can stay in their respective method folders while still appearing in the sidebar, benefiting both developers and users. It's particularly useful for large projects where documentation is critical but often neglected due to maintenance difficulties.

### Do you have examples of this feature in other projects?

- https://github.com/effector/patronum/tree/75ad9600ace84e61bba3428de75ac6695210b801/src/delay

### Participation

- [x] I am willing to submit a pull request for this feature.

**Comment by delucis** [comment]
Thanks for the feedback @sergeysova! We'll definitely be looking to add content layer support for when it is stable.

**Reply by sergeysova** [reply]
https://astro.build/blog/astro-5-beta/

Stable in Beta!

**Reply by sgalcheung** [reply]
When I tried to upgrade to the Astro 5.0 beta version, I received a mistake:

> Package subpath './jsx/renderer.js' is not defined by "exports" in /Users/sgalcheung/***/node_modules/.pnpm/@astrojs+mdx@3.1.7_astro@5.0.0-beta.2_@types+node@22.7.3_rollup@4.22.4_typescript@5.6.2_/node_modules/astro/package.json imported from /Users/sgalcheung/***/node_modules/.pnpm/@astrojs+mdx@3.1.7_astro@5.0.0-beta.2_@types+node@22.7.3_rollup@4.22.4_typescript@5.6.2_/node_modules/@astrojs/mdx/dist/index.js

The Astro upgrade some dependency, https://5-0-0-beta.docs.astro.build/en/guides/upgrade-to/v5/#astrojsmdx

> In Astro v4.x, Astro performed internal JSX handling for the @astrojs/mdx integration.
> 
> Astro v5.0 moves this responsibility to handle and render JSX and MDX to the @astrojs/mdx package directly. This means that Astro 5.0 is no longer compatible with older versions of the MDX integration.
> If your project includes .mdx files, you must upgrade @astrojs/mdx to the latest version (v4.0.0) so that your JSX can be handled properly by the integration.

But I can't override @astrojs/mdx
```
"overrides": {
    "@astrojs/starlight": {
      "@astrojs/mdx": "4.0.0-beta.1",
      "astro": "5.0.0-beta.2"
    }
  }
```
Should we make a beta version for this condition? Or have any other problem?

**Reply by sgalcheung** [reply]
```
❯ pnpm add @astrojs/mdx@4.0.0-beta.1 --save-exact

"pnpm": {
    "overrides": {
      "@astrojs/starlight>@astrojs/mdx": "4.0.0-beta.1"
    }
  }
```

**Reply by BANOnotIT** [reply]
Is there any way to add support for such with plugins for example?
